### PR TITLE
Change response error type from `String` to dedicated enum.

### PR DESCRIPTION
**Pull request by frewsxcv**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/10731)

### CODE REVIEW

**Comment by frewsxcv** [comment]
Definitely room for cleanup, but looking for earlier feedback.

File: components/net_traits/response.rs [hunk 1]
```diff
@@ -72,6 +75,55 @@ pub enum ResponseMsg {
     Errored
 }
 
+#[derive(Clone, Debug, Deserialize, Serialize, HeapSizeOf)]
+pub enum ResponseError {
+    ConnectionError(String),  // FIMXE: don't use string
```
Comment (frewsxcv) [inline components/net_traits/response.rs]: I'd love to make this `ConnectionError(std::io::Error)`, but can't seem to get it to work right. Any other ideas here?

**Comment by frewsxcv** [comment]
Unit tests aren't going to compile (I forgot to update them). But I'm going to wait for feedback before putting anymore effort into this.

**Comment by Ms2ger** [comment]
This seems like an improvement

**Comment by frewsxcv** [comment]
```
9:58 AM <frewsxcv> jdm: you around? do you think I should close https://github.com/servo/servo/pull/10731 ?
10:00 AM <•jdm> frewsxcv: yeah, I think it will make #9942 more annoying to rebase :(
```

https://github.com/servo/servo/pull/9942

**Pull request closed without merging**